api.js

```diff
@@ -44,19 +45,36 @@ const pullDeleting = (prop, obj) => {
   return v;
 };
 
-const makeBodiedRequest = (req, bodyProp, body, auth) =>
-  extractBody(
-    splitBodyProps(body).then(({ left: arr, right: nonArr }) =>
-      // console.error(arr, nonArr),
-      Object.keys(arr).reduce(
-        (req, k) => (applyArrayParams(k, arr[k], req), req),
-        req.use(log).field({ ...fixParams(nonArr)(pullDeleting('asPropOf', nonArr)), ...auth }),
-      ),
-    ),
-  )(bodyProp);
+const sleep = (exports.sleep = (seconds = 0.75) => {
+  const ms = seconds * 1000;
+  const start = Date.now();
+  while (Date.now() - start < ms) {}
+});
+
+// use to avoid tripping the discourse request limiter
+const sleepAsync = seconds =>
```

Comment:
Obviously this is where we'll need to talk about implementing a more sophisticated version of this, like I said in your other PR, we use this library for the dispute-tools to talk to discourse and don't want this kind of slowdown every time.

(Take the next part, or leave it, just a style nitpick)
And I know you probably just copied this from the discourse-node-api repository, but, from a stylistic perspective I'd love to see this be a function `sleeping` with the type: `Function<Promise<T>> -> Function<Promise<T>>` that just decorates the passed in function with whatever we come up with for rate-limiting. Then we can also easily add configuration flags for enabling or disabling rate limiting. It'd be something like this:

```javascript
const sleeping = (toSleepBefore, timeToSleep = defaultSleepSeconds) =>
  rateLimitingPreventionEnabled
    ? (...args) => doSleep(timeToSleep).then(() => toSleepBefore(...args))
    : toSleepBefore
```

Reply:
Check out what I ended up doing!